packages/react-aria-components/src/HiddenDateInput.tsx

```diff
@@ -66,7 +66,12 @@ export function useHiddenDateInput(props: HiddenDateInputProps, state: DateField
     inputStep = 3600; 
   }
 
-  let dateValue = state.value == null ? '' : state.value.toString();
+  let dateValue = '';
+  if (state.value != null && 'toAbsoluteString' in state.value) {
+    dateValue = state.value.toAbsoluteString().replace('Z', '');
```

Comment:
there will never be something after the Z?

Reply:
I think `toAbsoluteString` would not quite be right, because it returns the date in UTC whereas `datetime-local` expects it to be in the user's local time zone.

Also for granularity="day", the value may still be a CalendarDateTime or ZonedDateTime, we only display the date. Formatting it as a string will produce a full date time where the browser will expect only a date.

Maybe what we should do is always convert it to a CalendarDate or CalendarDateTime before converting to a string, depending on the granularity.

```tsx
dateValue = state.granularity === 'day' 
  ? toCalendarDate(state.value).toString() 
  : toCalendarDateTime('timeZone' in state.value ? toLocalTimeZone(state.value) : state.value).toString()
```